syntax = "proto3";

message ProduceRequest {
    string channel_id = 1;
    string message = 2;
}

message ProduceResponse {
    bool success = 1;
}

message ConsumeRequest {
    string channel_id = 1;
}

message ConsumeResponse {
    string message = 1;
}

message HealthCheckRequest { }

message HealthCheckResponse { 
    bool healthy = 1;
}

service ChannelService {
    rpc Produce(ProduceRequest) returns (ProduceResponse);
    rpc Consume(ConsumeRequest) returns (ConsumeResponse);
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}